### DISCUSSION TITLE
github.event.pull_request.base.sha not refering to base branching commit

### DISCUSSION THREAD

**Discussion by jeetee** [community/community#39880] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

As discovered in https://github.com/musescore/MuseScore/pull/13992#issuecomment-1325658661 and seemingly by someone else in https://github.com/actions/runner/issues/1689 it seems that there is no way to get the actual base commit reference for a pull request.

While I can see the need for having a reference to the *current head* of the base branch, I would not expect this to be the value of `github.event.pull_request.base.sha`, but rather of an alternative (to be added?) property such as `github.event.pull_request.base.head.sha` or the likes.

As it stands, it seems to me that there is no way to have an easy test of only the PR changes (in our case, a visual reference test that should run between branch base reference and PR's HEAD).
Unless I'm somehow missing the obvious?

**Comment by ossanna16** [comment]
Hi there @jeetee and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by jeetee** [comment]
I guess what I'm hoping to find was something along the lines of `git merge-base --fork-point <PR-merge-target>`

**Comment by VanTanev** [comment]
I want to confirm that the issue exists.

We have a PR 2 that references PR 1's branch as its base branch. It is correctly shown in the github interface.

However, in github actions, `github.event.pull_request.base.sha` gives us the last commit on the repository main branch instead PR 1's latest commit.

**Reply by vol4onok** [reply]
I have the same issue. `github.event.pull_request.base.sha` gives another commit instead of `git merge-base  origin/currentbranch origin/master`.

git merge-base  origin/currentbranch origin/master - correct sha
github.event.pull_request.base.sha - The last commit

**Reply by magos1983** [reply]
same problem for me

**Reply by ovflowd** [reply]
> However, in github actions, github.event.pull_request.base.sha gives us the last commit on the repository main branch instead PR 1's latest commit

You're looking for `github.event.pull_request.head.sha`

**Reply by brun0xff** [reply]
I'm facing the same problem here

**Comment by A132770** [comment]
I ran into this as well. I am trying to implement a type of status check that fails if a branch is not 'up-to-date' with main, to prevent subsequent more expensive status checks from running until main is merged in (GitHub's update feature does not prevent status checks from running).

**Comment by chrisconfirm** [comment]
anyone have an answer to this? i want to use pretty-quick, and have the same problem. i can not easily get the actual base commit reference. pretty-quick is therefore not finding any changes and not formatting any files. see https://github.com/azz/pretty-quick/issues/127

**Comment by Cowa** [comment]
I ran into this issue too. I'm using it for code coverage diff reports.

**Comment by carlescufi** [comment]
Please see https://github.com/orgs/community/discussions/59677#discussioncomment-10782359
